src/olympia/addons/models.py

```diff
@@ -1129,7 +1129,9 @@ def can_request_review(self):
         latest_version = self.find_latest_version(
             amo.RELEASE_CHANNEL_LISTED, exclude=(amo.STATUS_BETA,))
 
-        return latest_version is not None and latest_version.files.exists()
+        return (latest_version is not None and
+                not latest_version.is_webextension and
+                latest_version.files.exists())
```

Comment:
>  So, the issue conversation was misleading then

Not quite. The proper solution would be to check whether a version was rejected or not. Since we can't do that, we'll just exclude all webextensions because of auto-approval.
That leaves legacy extensions, of which new submissions are entirely disallowed and update submissions should be covered by https://github.com/wagnerand/addons-server/blob/6b72aff448ee23dc42f2680ec0258e8aceadfcdc/src/olympia/addons/models.py#L1123-L1127 for the most part.
There is one uncovered edge-case where a developer submits a legacy update for an add-on that has status `INCOMPLETE`, but I'm willing to live with that edge case, since that should happen quite rarely.

> Hmm, so this function is also used for the 'my submissions' list. Meaning anyone submitting a webextension will lose the ability to resume a submission.

Ugh, yeah. I grepped for `can_request_review` and saw the html file but failed to make the connection. So basically, this patch is useless as it's not (really) any better than my initial one :(